EnhancedMediaPlayer/MediaPlayerView.swift

```diff
@@ -0,0 +1,171 @@
+// MediaPlayerView.swift
+
+import SwiftUI
+
+public struct MediaPlayerView: View {
+    @ObservedObject var viewModel: MediaPlayerView.ViewModel
+    @State private var showActionButtons = false
+    
+    public init(viewModel: MediaPlayerView.ViewModel) {
+        self.viewModel = viewModel
+    }
+
+    public var body: some View {
+        GeometryReader { geometry in
+            ZStack {
+                AVPlayerView(mediaURL: viewModel.mediaURL, playerState: viewModel.playerState) {
+                    viewModel.onMediaFinished?()
+                }
+                
+                if showActionButtons {
+                    renderOverlay()
+                    renderActionButtons()
+                        .padding(.horizontal, geometry.size.width / Constants.paddingWidthDivider)
+                }
+            }
```

Comment:
We generally avoid using raw. `GeometryReaders` because they tend to break the layout. Try to bring the `readFrame`-like modifiers from the QS and use that instead